cmd/notation/registry.go

```diff
@@ -14,39 +15,52 @@ import (
 	"oras.land/oras-go/v2/registry/remote/auth"
 )
 
-func getSignatureRepository(opts *SecureFlagOpts, reference string) (notationregistry.Repository, error) {
+func getSignatureRepositoryClient(opts *SecureFlagOpts, reference string) (*remote.Repository, error) {
```

Comment:
All invocations of `getSignatureRepositoryClient` and `getRepositoryClient` are followed by a call of `notationRegistry.NewRepository`. That's a strong indicator that we should do refactor.

Reply:
Refactored it again.